sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeDisableAutomaticCheckForUpdateTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.appcenter.distribute;
+
+import com.microsoft.appcenter.http.HttpClient;
+import com.microsoft.appcenter.http.ServiceCallback;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static com.microsoft.appcenter.distribute.DistributeConstants.PREFERENCE_KEY_REQUEST_ID;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class DistributeDisableAutomaticCheckForUpdateTest extends AbstractDistributeTest {
+
+    @Test
+    public void disableAutomaticCheckForUpdateBeforeDistributeStartDoesNotCheckForUpdate() {
+
+        /* Disable automatic check for update then start. */
+        Distribute.disableAutomaticCheckForUpdate();
+        start();
+        Distribute.getInstance().onActivityResumed(mActivity);
+
+        /* No HTTP call done. */
+        verify(mHttpClient, never()).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), any(ServiceCallback.class));
+    }
+
+    @Test
+    public void disableAutomaticCheckForUpdateAfterDistributeStartChecksForUpdate() {
+
+        /* Start then call disable automatic check for update after Distribute has started. */
+        start();
+        Distribute.disableAutomaticCheckForUpdate();
+        Distribute.getInstance().onActivityResumed(mActivity);
+
+        /* HTTP call done. */
+        verify(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), any(ServiceCallback.class));
+    }
+
+    @Test
+    public void manualCheckWhileAutomaticCheckDisabledChecksForUpdate() {
+
+        /* Start then call disable automatic check for update after Distribute has started. */
+        Distribute.disableAutomaticCheckForUpdate();
+        start();
+        Distribute.checkForUpdate();
+        Distribute.getInstance().onActivityResumed(mActivity);
```

Comment:
Put verify right after `checkForUpdate` or comment why `onActivityResumed` is needed here. Also, please verify that there was no HTTP calls right before `checkForUpdate` call